client/client_test.go

```diff
@@ -3617,38 +3620,76 @@ func TestGetAllTargetInfo(t *testing.T) {
 	// level2 - signed by targets/level2 and targets/level1/level2, with the same hash
 
 	// Positive cases
-	targetData, err := repo.GetAllTargetMetadataByName("current")
-	require.NoError(t, err)
-	require.NotNil(t, targetData)
-	require.Len(t, targetData, 3)
-	require.Equal(t, targetData[data.CanonicalTargetsRole], *targetsCurrentTarget)
-	require.Equal(t, targetData["targets/level1"], *level1CurrentTarget)
-	require.Equal(t, targetData["targets/level2"], *level2CurrentTarget)
-
-	targetData, err = repo.GetAllTargetMetadataByName("other")
-	require.NoError(t, err)
-	require.NotNil(t, targetData)
-	require.Len(t, targetData, 1)
-	require.Equal(t, targetData["targets/level1"], *level1OtherTarget)
-
-	targetData, err = repo.GetAllTargetMetadataByName("latest")
-	require.NoError(t, err)
-	require.NotNil(t, targetData)
-	require.Len(t, targetData, 1)
-	require.Equal(t, targetData[data.CanonicalTargetsRole], *targetsLatestTarget)
-
-	targetData, err = repo.GetAllTargetMetadataByName("level2")
-	require.NoError(t, err)
-	require.NotNil(t, targetData)
-	require.Len(t, targetData, 2)
-	require.Equal(t, targetData["targets/level2"], *level2Level2Target)
-	require.Equal(t, targetData["targets/level1/level2"], *level1Level2Level2Target)
+	targetSignatureData, err := repo.GetAllTargetMetadataByName("current")
+	require.NoError(t, err)
+	require.NotNil(t, targetSignatureData)
+	require.Len(t, targetSignatureData, 3)
+	var makeSureWeHitEachCase int
+	for _, tarSigStr := range targetSignatureData {
+		switch tarSigStr.Role.Name {
+		case data.CanonicalTargetsRole:
+			require.Len(t, tarSigStr.Signatures, 1)
+			require.Equal(t, repo.CryptoService.ListKeys(data.CanonicalTargetsRole)[0], tarSigStr.Signatures[0].KeyID)
+			require.Equal(t, tarSigStr.Target, *targetsCurrentTarget)
+			makeSureWeHitEachCase = makeSureWeHitEachCase + 1
+		case "targets/level1":
+			require.Len(t, tarSigStr.Signatures, 1)
+			require.Equal(t, key1.ID(), tarSigStr.Signatures[0].KeyID)
+			require.Equal(t, tarSigStr.Target, *level1CurrentTarget)
+			makeSureWeHitEachCase = makeSureWeHitEachCase + 2
+		case "targets/level2":
+			require.Len(t, tarSigStr.Signatures, 1)
+			require.Equal(t, key2.ID(), tarSigStr.Signatures[0].KeyID)
+			require.Equal(t, tarSigStr.Target, *level2CurrentTarget)
+			makeSureWeHitEachCase = makeSureWeHitEachCase + 4
```

Comment:
nit: not sure if it's worth keeping this as a loop instead of just indexing by the role name